pkg/jobs/hive/hive_test.go

```diff
@@ -776,457 +797,457 @@ func TestUpgradeCluster(t *testing.T) {
-				"cluster.open-cluster-management.io/upgrade-allow-not-recommended-versions": "true",
-			},
-		},
-		Spec: clustercuratorv1.ClusterCuratorSpec{
-			DesiredCuration: "upgrade",
-			Upgrade: clustercuratorv1.UpgradeHooks{
-				DesiredUpdate: "4.5.14",
-			},
-		},
-	}
-
-	currentClusterVersion := &clusterversionv1.ClusterVersion{
-		TypeMeta: v1.TypeMeta{
-			APIVersion: "config.openshift.io/v1",
-			Kind:       "ClusterVersion",
-		},
-		ObjectMeta: v1.ObjectMeta{
-			Name: "version",
-		},
-		Spec: clusterversionv1.ClusterVersionSpec{
-			Channel:   "stable-4.5",
-			ClusterID: "201ad26c-67d6-416a",
-			Upstream:  "https://api.openshift.com/api",
-		},
-		Status: clusterversionv1.ClusterVersionStatus{
-			AvailableUpdates: []clusterversionv1.Update{
-				{
-					Version: "4.5.10",
-					Image:   "quay.io/openshift-release-dev/ocp-release@sha256:71e158c6173ad6aa6e356c119a87459196bbe70e89c0db1e35c1f63a87d90676",
-				},
-			},
-		},
-	}
-
-	b, _ := json.Marshal(currentClusterVersion)
-
-	client := clientfake.NewFakeClientWithScheme(s, []runtime.Object{
-		clustercurator, getManagedClusterInfo(),
-	}...)
-
-	go func() {
-		for i := 0; i < 10; i++ {
-			time.Sleep(1 * time.Second)
-			resultmcview := managedclusterviewv1beta1.ManagedClusterView{}
-			err := client.Get(context.TODO(), types.NamespacedName{
-				Namespace: ClusterName,
-				Name:      ClusterName,
-			}, &resultmcview)
-			if err != nil {
-				klog.Error("failed to get managedClusterview.", err)
-				continue
-			}
-			resultmcview.Status.Result.Raw = b
-			err = client.Update(context.TODO(), &resultmcview)
-			if err != nil {
-				klog.Error("failed to update managedClusterview.", err)
-				continue
-			}
-			updatedresultmcview := managedclusterviewv1beta1.ManagedClusterView{}
-			err = client.Get(context.TODO(), types.NamespacedName{
-				Namespace: ClusterName,
-				Name:      ClusterName,
-			}, &updatedresultmcview)
-			if err != nil {
-				klog.Error("failed to get managedClusterview.", err)
-				continue
-			}
-			if updatedresultmcview.Status.Result.Raw != nil {
-				break
-			}
-		}
-	}()
-
-	go func() {
-		for i := 0; i < 10; i++ {
-			time.Sleep(1 * time.Second)
-			resultmca := managedclusteractionv1beta1.ManagedClusterAction{}
-			err := client.Get(context.TODO(), types.NamespacedName{
-				Namespace: ClusterName,
-				Name:      ClusterName,
-			}, &resultmca)
-
-			if err == nil {
-				patch := []byte(`{"status":{"conditions":[
-							{
-								"lastTransitionTime": "2021-04-28T16:19:38Z",
-								"message": " Resource action is done.",
-								"reason": "ActionDone",
-								"status": "True",
-								"type": "Completed"
-							}]}}`)
-				client.Patch(context.Background(), &resultmca, clientv1.RawPatch(types.MergePatchType, patch))
-				break
-			}
-		}
-	}()
-
-	assert.Nil(t, UpgradeCluster(client, ClusterName, clustercurator), "Upgrade started successfully to non-recommended version")
-}
+// func TestUpgradeClusterWithChannelUpstream(t *testing.T) {
```

Comment:
@fxiang1 I was able to finally get through everything except for these 4 tests in hive_test.go related to cluster curator upgrade that don't rely on the hive client anyways. If the rest of the PR is okay, would you be okay to rebase your branch with the changes and help take a look at those unit tests?

Reply:
Sure I can take a look. Thanks David!